### PR TITLE
Fix special HOC event review page

### PR DESCRIPTION
**Pull request by islemaster**
Fix up the tool used for reviewing special Hour of Code events to remove a SOLR query, update everything for 2018, and fix up some query performance issues.

### CODE REVIEW

**Comment by islemaster** [comment]
@tanyaparker I've still got to write some tests for this, but it should be fully functional at this point. Can you take an initial look and let me know what you think?

File: pegasus/sites.v3/code.org/public/teacher-dashboard/review-hociyskvuwa/splat.haml [hunk 1]
```diff
@@ -98,7 +89,7 @@ title: Review Hour of Code special events
 
   function submitReview(secret, review, row) {
     $.ajax({
-      url: "/v2/forms/HocSignup2017/" + secret + "/review",
+      url: "/v2/forms/HocSignup2018/" + secret + "/review",
```
Comment (tanyaparker) [inline pegasus/sites.v3/code.org/public/teacher-dashboard/review-hociyskvuwa/splat.haml]: We use DCDO `hoc_year` so we don't have to remember all the places to update. :)

https://github.com/code-dot-org/code-dot-org/blob/0cd6428b533faa05ac4c95d89bf09ffd4bb6aba0/pegasus/sites.v3/hourofcode.com/public/events/organizations.haml#L8

**Comment by islemaster** [comment]
Tests added!  I believe this is ready for a full review.  Thanks!

File: pegasus/helper_modules/hoc_event_review.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      query = query.where(STATE_CODE_COLUMN => state.upcase)
+    elsif country
+      query = query.where(COUNTRY_CODE_COLUMN => country.upcase)
+    end
+
+    query
+  end
+
+  def self.kind
+    "HocSignup#{DCDO.get('hoc_year', 2017)}"
```
Comment (Erin007) [inline pegasus/helper_modules/hoc_event_review.rb:79]: 2017?

File: pegasus/sites.v3/code.org/public/teacher-dashboard/review-hociyskvuwa/index.haml [hunk 3]
```diff
@@ -21,40 +20,18 @@ title: Review Hour of Code special events
...
-  -by_country[country_code] << event
-  -country_names[country_code] ||= event['location_country_s'].to_s
-
-  -if country_code == 'US'
-    -state_code = event['location_state_code_s'].to_s
-    -by_state[state_code] ||= []
-    -by_state[state_code] << event
-    -state_names[state_code] ||= event['location_state_s'].to_s
-
 %ul
```
Comment (Erin007) [inline pegasus/sites.v3/code.org/public/teacher-dashboard/review-hociyskvuwa/index.haml:23]: yay! look at that giant chunk of deleted haml and SOLR query!

File: pegasus/helper_modules/hoc_event_review.rb (hunk 2 above)
Comment (islemaster) [reply pegasus/helper_modules/hoc_event_review.rb:79]: :man_shrugging: It's the default we use elsewhere when `hoc_year` isn't set - I believe the current expectation is that we'll _always_ have one of these set on production, though.

**Pull request merged**